configure

```diff
@@ -0,0 +1,384 @@
+                                  It's passed after TEST_CXXFLAGS without overwrite it.
+  CPPFLAGS                      If you want to use some arguments in C++ compiler preprocessor.
+  LDFLAGS                       If you want to use some arguments in Linker command line.
+  LDFLAGS_EXTRA                 If you want to use some arguments in Linker command line.
+                                  It's passed after LDFLAGS without overwrite it.
+  TEST_LDFLAGS                  If you want to use some arguments in Linker for testing.
+  TEST_LDFLAGS_EXTRA            If you want to use some arguments in Linker for testing.
+                                  It's passed after TEST_LDFLAGS without overwrite it.
+
+Note: be careful about passing a path with space as prefix/eprefix/sbindir/mandir:
+for this thing You need to use backslash (\) with double quotation (") for every space.
+For example: ./configure --sbindir="/usr/bin\ with\ space/"
+
+See BUILD file for more information.
+EOF
+    exit 0
+}
+
+## Flag parser
+for arg in "$@"; do
+    case "$arg" in
+        --help) usage ;;
+        --quiet)
+            info() { true; }
+            sub_info() { true; }
+            warning() { true; }
+            sub_warning() { true; }
+        ;;
+        --clean) rm -f test* & rm -f mconfig && exit 0 ;;
+        --platform=*) PLATFORM="${arg#*=}" && shift ;;
+        --prefix=*) PREFIX="${arg#*=}" && shift ;;
+        --exec-prefix=*) EPREFIX="${arg#*=}" && shift;;
+        --sbindir=*) SBINDIR="${arg#*=}" && shift ;;
+        --mandir=*) MANDIR="${arg#*=}" && shift ;;
+        --syscontrolsocket=*) SYSCONTROLSOCKET="${arg#*=}" && shift ;;
+        --shutdown-prefix=*) SHUTDOWN_PREFIX="${arg#*=}" && shift ;;
+        --enable-shutdown|--enable-shutdown=yes) BUILD_SHUTDOWN=yes ;;
+        --disable-shutdown|--enable-shutdown=no) BUILD_SHUTDOWN=no ;;
+        --enable-cgroups|--enable-cgroups=yes) SUPPORT_CGROUPS=1 ;;
+        --disable-cgroups|--enable-cgroups=no) SUPPORT_CGROUPS=0 ;;
+        --enable-sanitize|--enable-sanitize=yes) NO_SANITIZE=0 ;;
+        --disable-sanitize|--enable-sanitize=no) NO_SANITIZE=1 ;;
+        --enable-utmpx|--enable-utmpx=yes) USE_UTMPX=1 ;;
+        --disable-utmpx|--enable-utmpx=no) USE_UTMPX=0 ;;
+        CXX=*|CXX_FOR_BUILD=*|CXXFLAGS_FOR_BUILD=*|CPPFLAGS_FOR_BUILD=*\
+        |LDFLAGS_FOR_BUILD=*|CXXFLAGS=*|CXXFLAGS_EXTRA=*|TEST_CXXFLAGS=*\
+        |TEST_CXXFLAGS_EXTRA|LDFLAGS=*|LDFLAGS_EXTRA=*|TEST_LDFLAGS=*\
+        |TEST_LDFLAGS_EXTRA=*|CPPFLAGS=*)
+            envvar=$(expr "x$arg" : 'x\([^=]*\)=')
+            envvar_value=$(expr "X$arg" : '[^=]*=\(.*\)') || envvar_value="" # Fix empty var bug
+            eval "$envvar=\$envvar_value"
+        ;;
+        *=*) warning Unknown variable: "$(expr "x$arg" : 'x\([^=]*\)=')" ;;
+        *) warning Unknown argument: "$arg" ;;
+    esac
+done
+
+## General Defines
+: "${PLATFORM:=$(uname)}"
```

Comment:
Now `configure` supports all variables be imported from environment. It's a good behavior? Can this be a problem?

Reply:
I think it's ok as long as all environment variables are documented (eg in `--help` output)

Reply:
So it should be documented that PLATFORM, PREFIX, EPREFIX etc, take their values from the environment (and what this means). Or otherwise, don't take these values from the environment!

Reply:
So I think it's better to don't take values from environment.

Reply:
Yep, ok